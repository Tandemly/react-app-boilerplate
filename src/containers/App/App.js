// @flow
import React, { Component } from "react";
import Navbar from "../../components/Navbar";
import logo from "../../images/tandemly-full-logo-black-white.svg";
import Button from "../../components/Button";
import "./App.scss";

const App = () => (
  <div className="App">
    <Navbar
      transparent
      renderLogo={() => (
        <a className="navbar-item" href="#">
          <img src={logo} alt="Tandem.ly Logo" width="112" height="28" />
        </a>
      )}
      renderLeftMenu={() => [
        <a href="#" className="navbar-item">
          One
        </a>,
        <a href="#" className="navbar-item">
          Two
        </a>,
        <a href="#" className="navbar-item is-active">
          Three
        </a>
      ]}
    />
    <div className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <h2>Welcome to React</h2>
    </div>
    <p className="App-intro">
      To get started, edit <code>src/App.js</code> and save to reload.
    </p>
    <Button text="Click Me" className="is-primary" />
  </div>
);

export default App;
