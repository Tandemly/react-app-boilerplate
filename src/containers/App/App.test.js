import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import App from "./App";

it("renders without crashing", () => {
  const app = shallow(<App />);
  expect(app).toMatchSnapshot();
});
