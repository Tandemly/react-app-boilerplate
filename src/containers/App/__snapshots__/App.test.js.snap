// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          renderBrand={[Function]}
          renderLeft={[Function]}
          transparent={true}
/>,
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="tandemly-full-logo-black-white.svg"
          />
          <h2>
                    Welcome to React
          </h2>
</div>,
        <p
          className="App-intro"
>
          To get started, edit 
          <code>
                    src/App.js
          </code>
           and save to reload.
</p>,
        <Styleable(Button)
          className="is-primary"
          css={Object {}}
          text="Click Me"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "renderBrand": [Function],
          "renderLeft": [Function],
          "transparent": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="tandemly-full-logo-black-white.svg"
/>,
            <h2>
              Welcome to React
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "tandemly-full-logo-black-white.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to React",
            },
            "ref": null,
            "rendered": "Welcome to React",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "To get started, edit ",
            <code>
              src/App.js
</code>,
            " and save to reload.",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "To get started, edit ",
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.js",
            },
            "ref": null,
            "rendered": "src/App.js",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "className": "is-primary",
          "css": Object {},
          "text": "Click Me",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            renderBrand={[Function]}
            renderLeft={[Function]}
            transparent={true}
/>,
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="tandemly-full-logo-black-white.svg"
            />
            <h2>
                        Welcome to React
            </h2>
</div>,
          <p
            className="App-intro"
>
            To get started, edit 
            <code>
                        src/App.js
            </code>
             and save to reload.
</p>,
          <Styleable(Button)
            className="is-primary"
            css={Object {}}
            text="Click Me"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "renderBrand": [Function],
            "renderLeft": [Function],
            "transparent": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="tandemly-full-logo-black-white.svg"
/>,
              <h2>
                Welcome to React
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "tandemly-full-logo-black-white.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to React",
              },
              "ref": null,
              "rendered": "Welcome to React",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.js
</code>,
              " and save to reload.",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.js",
              },
              "ref": null,
              "rendered": "src/App.js",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "className": "is-primary",
            "css": Object {},
            "text": "Click Me",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
